src/stream/src/common/builder.rs

```diff
@@ -52,15 +52,22 @@ pub struct StreamChunkBuilder {
     size: usize,
 }
 
+impl Drop for StreamChunkBuilder {
+    fn drop(&mut self) {
+        assert_eq!(self.size, 0, "dropping non-empty stream chunk builder");
+    }
+}
+
 impl StreamChunkBuilder {
     pub fn new(
         capacity: usize,
         data_types: &[DataType],
         update_start_pos: usize,
         matched_start_pos: usize,
     ) -> Result<Self> {
-        // TODO: assert `capacity != 0`
-        let ops = Vec::with_capacity(capacity);
+        assert!(capacity > 1);
+        let capacity = capacity - 1;
```

Comment:
Why `-1` here? Is there any comment?

Reply:
Added comment for this.

For example, when `capacity = 3`, we generate a chunk when `size = capacity - 1 = 2`.
However, when the last op is `UpdateDelete`, we leave a space for the upcoming `UpdateInsert`, after which we can generate a chunk without splitting `UpdateDelete/UpdateInsert` or violating capacity requirement.
```
capacity = 3:
------------------------------------
Delete -> Delete (Generate a chunk)
------------------------------------
Insert -> UpdateDelete -> ? (Do not generate a chunk)
Insert -> UpdateDelete -> UpdateInsert (Generate a chunk)
```